src/cache/mod.rs

```diff
@@ -122,3 +156,31 @@ impl Cache {
         self.driver.clear().await
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{app::AppContext, tests_cfg};
+
+    #[tokio::test]
+    async fn can_get_or_insert() {
+        let app_ctx = tests_cfg::app::get_app_context().await;
+        let get_key = "loco";
+
+        assert_eq!(app_ctx.cache.get(get_key).await.unwrap(), None);
+
+        let result = app_ctx
+            .cache
+            .get_or_insert(get_key, &app_ctx, |_ctx: &AppContext| {
```

Comment:
I assume there is no way to capture `ctx` right? (this is why its passed to the function)
If possible to get something like this:

```
.get_or_insert("key", async || {
     Users.find(..&ctx).await
})
```

this would be the most realistic. notice that 

1) `F` must be async because to be realistic people would probably make async db calls there
2) `ctx` if possible, should be captured to avoid passing it as a parameter just to "ship it" to `F`

Capturing `ctx` would probably require some concept of `sync` and `send`, so might require wrapping `ctx` in `Arc` and/or `Mutex`